commit 1b0975ee3bdd3eb19a47371c26fd7ef8f7f6b599
Author: Mark Brown <broonie@kernel.org>
Date:   Wed Jul 12 21:23:17 2023 +0100

    kunit: qemu_configs: Enable all architectural features for arm64
    
    While it probably doesn't make a huge difference given the current KUnit
    coverage we will get the best coverage of arm64 architecture features if
    we specify -cpu=max rather than picking a specific CPU, this will include
    all architecture features that qemu supports including many which have not
    yet made it into physical implementations.
    
    Due to performance issues emulating the architected pointer authentication
    algorithm it is recommended to use the implementation defined algorithm
    that qemu has instead, this should make no meaningful difference to the
    coverage and will run the tests faster.
    
    Signed-off-by: Mark Brown <broonie@kernel.org>
    Reviewed-by: David Gow <davidgow@google.com>
    Signed-off-by: Shuah Khan <skhan@linuxfoundation.org>

diff --git a/tools/testing/kunit/qemu_configs/arm64.py b/tools/testing/kunit/qemu_configs/arm64.py
index 67d04064f785..d3ff27024755 100644
--- a/tools/testing/kunit/qemu_configs/arm64.py
+++ b/tools/testing/kunit/qemu_configs/arm64.py
@@ -9,4 +9,4 @@ CONFIG_SERIAL_AMBA_PL011_CONSOLE=y''',
 			   qemu_arch='aarch64',
 			   kernel_path='arch/arm64/boot/Image.gz',
 			   kernel_command_line='console=ttyAMA0',
-			   extra_qemu_params=['-machine', 'virt', '-cpu', 'cortex-a57'])
+			   extra_qemu_params=['-machine', 'virt', '-cpu', 'max,pauth-impdef=on'])